crates/ruff_workspace/src/pyproject.rs

```diff
@@ -99,28 +101,31 @@ pub fn find_settings_toml<P: AsRef<Path>>(path: P) -> Result<Option<PathBuf>> {
 /// Find the path to the user-specific `pyproject.toml` or `ruff.toml`, if it
 /// exists.
 pub fn find_user_settings_toml() -> Option<PathBuf> {
-    // Search for a user-specific `.ruff.toml`.
-    let mut path = dirs::config_dir()?;
-    path.push("ruff");
-    path.push(".ruff.toml");
-    if path.is_file() {
-        return Some(path);
-    }
-
-    // Search for a user-specific `ruff.toml`.
-    let mut path = dirs::config_dir()?;
-    path.push("ruff");
-    path.push("ruff.toml");
-    if path.is_file() {
-        return Some(path);
+    let strategy = etcetera::base_strategy::choose_base_strategy().ok()?;
+    let config_dir = strategy.config_dir().join("ruff");
+
+    // Search for a user-specific `.ruff.toml`, then a `ruff.toml`, then a `pyproject.toml`.
+    for filename in [".ruff.toml", "ruff.toml", "pyproject.toml"] {
+        let path = config_dir.join(filename);
+        if path.is_file() {
+            return Some(path);
+        }
     }
 
-    // Search for a user-specific `pyproject.toml`.
-    let mut path = dirs::config_dir()?;
-    path.push("ruff");
-    path.push("pyproject.toml");
-    if path.is_file() {
-        return Some(path);
+    // On macOS, we used to support reading from `/Users/Alice/Library/Application Support`.
+    if cfg!(target_os = "macos") {
+        let deprecated_config_dir = dirs::config_dir()?.join("ruff");
```

Comment:
Could we use https://docs.rs/etcetera/latest/etcetera/app_strategy/struct.Apple.html here and remove the `dirs` dependency (trading one dependency for another ;))

Reply:
Okay I checked, we can use the `Apple` strategy but we need to use the `data_dir` to map to the `Application Support` folder (hardcoded). 

The `etcetera` crate clearly documents that `config_dir` shouldn't be used on Mac when the config is user editable.

Reply:
Interesting. Feel free to merge.